File: pdfsecondmonitor/CtlImage.vb
Public Class CtlImage

    Private _dispacher As FormDispacher = FormDispacher.GetInstance

    Private _pictureBox As PictureBox

    Private Sub Rotate(flip As RotateFlipType)
        Dim bmp As Bitmap = New Bitmap(_fileViewParam.FileName)
        bmp.RotateFlip(flip)
        _pictureBox.Image = bmp
    End Sub
    Private Sub None()
        _pictureBox.ImageLocation = _fileViewParam.FileName
    End Sub
    Private Sub btnRotate_Click(sender As Object, e As EventArgs) Handles btnRotate180.Click
        Rotate(RotateFlipType.Rotate180FlipNone)
    End Sub

    Private Sub btnRotate90_Click(sender As Object, e As EventArgs) Handles btnRotate90.Click
        Rotate(RotateFlipType.Rotate90FlipNone)
    End Sub

    Private Sub btnRotate0_Click(sender As Object, e As EventArgs) Handles btnRotate0.Click
        None()
    End Sub

    Private Sub btnRotate270_Click(sender As Object, e As EventArgs) Handles btnRotateM90.Click
        Rotate(RotateFlipType.Rotate270FlipNone)
    End Sub

    Private _fileViewParam As FileViewParam

    Public Sub SetFileInfo(f As FileViewParam)
        Me._fileViewParam = f
        If f Is Nothing Then
            Return
        End If
        _pictureBox = _dispacher.ShowImage()
        None()
        _pictureBox.SizeMode = PictureBoxSizeMode.Zoom 'サイズ調整


    End Sub

    Public Sub ControlEnabled()

        Me.Enabled = Not (_fileViewParam Is Nothing)

    End Sub

End Class

File: MediaToSecondMonitor/ctlImage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace MediaToSecondMonitor
{
    public partial class ctlImage : UserControl
    {
        public ctlImage()
        {
            InitializeComponent();
        }
        private PictureBox _pictureBox;

        private void Rotate(RotateFlipType flip)
        {
            Bitmap bmp = new Bitmap(_fileViewParam.FileName);
            bmp.RotateFlip(flip);
            _pictureBox.Image = bmp;
        }
        private void None()
        {
            _pictureBox.ImageLocation = _fileViewParam.FileName;
        }
        private void btnRotate_Click(object sender, EventArgs e)
        {
            Rotate(RotateFlipType.Rotate180FlipNone);
        }

        private void btnRotate90_Click(object sender, EventArgs e)
        {
            Rotate(RotateFlipType.Rotate90FlipNone);
        }

        private void btnRotate0_Click(object sender, EventArgs e)
        {
            None();
        }

        private void btnRotate270_Click(object sender, EventArgs e)
        {
            Rotate(RotateFlipType.Rotate270FlipNone);
        }

        private FormDispacher _dispacher = FormDispacher.Instance;
        private FileViewParam _fileViewParam;

        public void SetFileInfo(FileViewParam f)
        {
            this._fileViewParam = f;
            if (f == null)
                return;
            _pictureBox = _dispacher.GetPictureBox();
            None();
            _pictureBox.SizeMode = PictureBoxSizeMode.Zoom; // サイズ調整
        }

        public void ControlEnabled()
        {
            this.Enabled = !(_fileViewParam == null);
        }
    }
}
